File: egghead.io/data-structures-and-algorithms-in-js/lesson-02-queue-data-structure-in-js/custom_queue.py
class CustomQueue:
  def __init__(self):
    self.queue = [];

  def enqueue(self, item):
    self.queue.insert(0, item)

  def dequeue(self):
    self.queue.pop()

  def peek(self):
    return self.queue[self.size() - 1]

  def size(self):
    return len(self.queue)

  def is_empty(self):
    return self.size() == 0

File: egghead.io/data-structures-and-algorithms-in-js/lesson-02-queue-data-structure-in-js/custom-queue.js
class CustomQueue {
  constructor() { 
    this.queue = []; 
  }

  enqueue(item) { 
    this.queue.unshift(item); 
  }

  dequeue() { 
    return this.queue.pop(); 
  }

  peek() { 
    return this.queue[this.size - 1]; 
  }

  get size() { 
    return this.queue.length;
  }

  isEmpty() { 
    return this.size === 0; 
  }
}

module.exports = CustomQueue;
